swagger-templates/csharp/ApiClient.mustache

```diff
@@ -112,7 +112,16 @@ namespace {{packageName}}.Client
 
             // add header parameter, if any
             foreach(var param in headerParams)
-                request.AddHeader(param.Key, param.Value);
+            {
+                if (param.Key.Equals("Authorization"))
```

Comment:
The Note in he ms documentation talking about string comparison
https://msdn.microsoft.com/en-us/library/cc165449.aspx